pkg/proxy/proxy.go

```diff
@@ -52,14 +54,34 @@ func singleJoiningSlash(a, b string) string {
 }
 
 // AddHost undocumented
-func (p *MultiHostReverseProxy) AddHost(policy string, target *url.URL, endpoint string) {
+func (p *MultiHostReverseProxy) AddHost(policy string, target *url.URL, rt config.Route) {
 	targetQuery := target.RawQuery
 	if p.Directors[policy] == nil {
 		p.Directors[policy] = make(map[string]func(req *http.Request))
 	}
-	p.Directors[policy][endpoint] = func(req *http.Request) {
+	p.Directors[policy][rt.Endpoint] = func(req *http.Request) {
 		req.URL.Scheme = target.Scheme
 		req.URL.Host = target.Host
+		if rt.ApacheVHost {
+			req.Host = target.Host
+
+			// force http 1.1
+			nohttp2 := &http.Transport{
```

Comment:
I'd assign `p.Transport := &http.Transport{...}` and save an allocation

Reply:
I tried without this and it still works ... no bad request.

Reply:
🗑 then with this code :D